src/stdlib/memcpy.mbt

```diff
@@ -4,3 +4,26 @@ pub extern "c" fn memcpy(
   src : @c.Pointer[Byte],
   len : UInt64,
 ) -> Unit = "moonbit_c_memcpy"
+
+///|
+fn bytes_to_c_ptr(bytes : Bytes) -> @c.Pointer[Byte] = "%identity"
+
+///|
+pub fn memcpy_from_bytes(
+  dest : @c.Pointer[Byte],
+  source : Bytes,
+  length : UInt64,
+) -> Unit {
```

Comment:
If I'm correct about the memory leak, I will probably suggest the implementation of this function to be simply `= "moonbit_c_memcpy"`, and add `#borrow(source)` to tell the compiler not to increment the `Bytes` when passed as the `source` parameter. I believe this attribute is only available to `extern "c"` function.